src/gui/qgslegendfilterbutton.h

```diff
@@ -0,0 +1,85 @@
+/***************************************************************************
+    qgslegendfilterbutton.h - QToolButton for legend filter by map content
+     --------------------------------------
+    Date                 : June 2015
+    Copyright            : (C) 2015 by Hugo Mercier at Oslandia
+    Email                : hugo dot mercier at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGS_LEGEND_FILTER_BUTTON_H
+#define QGS_LEGEND_FILTER_BUTTON_H
+
+#include <QToolButton>
+
+class QgsVectorLayer;
+
+/** \ingroup gui
+ * \class QgsFilterLegendButton
+ * A tool button that allows to enable or disable legend filter by contents of the map.
+ * An additional pop down menu allows to define a boolean expression to refine the filtering.
+ * @note added in 2.12
+ */
+
+class GUI_EXPORT QgsLegendFilterButton: public QToolButton
+{
+    Q_OBJECT
+
+  public:
+    /**
+     * Construct a new filter legend button
+     *
+     * @param parent The parent QWidget
+     */
+    QgsLegendFilterButton( QWidget* parent = 0 );
+    ~QgsLegendFilterButton();
+
+    /*
```

Comment:
@mhugo these should be /** , or it's just a regular c++ comment. I think that's why the doc coverage test is complaining still